internal/assets/manager.go

```diff
@@ -189,6 +218,11 @@ func (am *assetManager) GetTokenAccounts(ctx context.Context, ns, typeName, name
 	return am.database.GetTokenAccounts(ctx, filter.Condition(filter.Builder().Eq("protocolid", pool.ProtocolID)))
 }
 
+func (am *assetManager) ValidateTokenPoolTx(ctx context.Context, pool *fftypes.TokenPool, protocolTxID string) error {
+	// TODO: validate that the given token pool was created with the given protocolTxId
```

Comment:
I attached a very high level picture of the flow I'm envisioning, but I've realized that I'm framing the question to the connector as "did Ethereum transaction 0xabc create pool F1", and the connector actually doesn't have the means to answer that.

We've hit this before, but ethconnect currently doesn't give a way to look up the events emitted from a particular transaction. I think it may be possible by decoding the output of `eth_getTransactionReceipt`, but I'm not quite sure how difficult it would be.

Reply:
This is great. Think it would be good to add them to https://github.com/hyperledger/firefly/issues/218

Reply:
> I think it may be possible by decoding the output of `eth_getTransactionReceipt`

The event stream code is processing these same binary logs, so it should certainly be possible to (just like you did for transaction inputs) allow the logs to be processed: https://github.com/hyperledger/firefly-ethconnect/blob/e708f38f5eac8ad2c4a6b084577c3954e75e7488/internal/events/logprocessor.go#L110-L181

The sticky bit will be making sure the code knows how to find the right event ABI. So the most practical and easy to implement version of the API, would probably be to add a `/retrieve` as a peer to `/subscribe` under the event on the REST ABI. A lot like you did for the method input parameters.

https://github.com/hyperledger/firefly-ethconnect/blob/8545f1c09d5fc9829508dac9a08d8c57210079eb/internal/contractgateway/rest2eth.go#L312

Reply:
Thanks for the tips - this will definitely be a work item that needs more exploration. Opened #222 to track it further.